modulemd/modulemd-eol.c

```diff
@@ -0,0 +1,1196 @@
+        error, MODULEMD_ERROR, MMD_ERROR_VALIDATE, "Eol message is unset.");
+      return FALSE;
+    }
+
+  return TRUE;
+}
+
+static void
+modulemd_eol_finalize (GObject *object)
+{
+  ModulemdEol *self = (ModulemdEol *)object;
+
+  g_clear_pointer (&self->module_name, g_free);
+  g_clear_pointer (&self->module_stream, g_free);
+  g_clear_pointer (&self->module_context, g_free);
+  g_clear_pointer (&self->message, g_free);
+  g_clear_pointer (&self->obsoleted_by_module_name, g_free);
+  g_clear_pointer (&self->obsoleted_by_module_stream, g_free);
+
+  G_OBJECT_CLASS (modulemd_eol_parent_class)->finalize (object);
+}
+
+guint64
+modulemd_eol_get_version (ModulemdEol *self)
+{
+  return self->version;
+}
+
+
+guint64
+modulemd_eol_get_modified (ModulemdEol *self)
+{
+  return self->modified;
+}
+
+
+gboolean
+modulemd_eol_get_reset (ModulemdEol *self)
+{
+  return self->reset;
+}
+
+
+const gchar *
+modulemd_eol_get_module_name (ModulemdEol *self)
+{
+  return self->module_name;
+}
+
+
+const gchar *
+modulemd_eol_get_module_stream (ModulemdEol *self)
+{
+  return self->module_stream;
+}
+
+
+const gchar *
+modulemd_eol_get_module_context (ModulemdEol *self)
+{
+  return self->module_context;
+}
+
+
+guint64
+modulemd_eol_get_eol_date (ModulemdEol *self)
+{
+  /* It should not be possible to set reset and eol_date
+   * at the same time but to be sure check anyway */
+  if (self->reset)
+    {
+      return 0;
+    }
+  return self->eol_date;
+}
+
+
+const gchar *
+modulemd_eol_get_message (ModulemdEol *self)
+{
+  return self->message;
+}
+
+
+const gchar *
+modulemd_eol_get_obsoleted_by_module_name (ModulemdEol *self)
+{
+  /* It should not be possible to set reset and obsoleted_by*
+   * at the same time but to be sure check anyway */
+  if (self->reset)
+    {
+      return NULL;
+    }
+  return self->obsoleted_by_module_name;
+}
+
+
+const gchar *
+modulemd_eol_get_obsoleted_by_module_stream (ModulemdEol *self)
+{
+  /* It should not be possible to set reset and obsoleted_by*
+   * at the same time but to be sure check anyway */
+  if (self->reset)
+    {
+      return NULL;
+    }
+  return self->obsoleted_by_module_stream;
+}
+
+
+static void
+modulemd_eol_set_version (ModulemdEol *self, guint64 version)
+{
+  g_return_if_fail (MODULEMD_IS_EOL (self));
+  g_return_if_fail (version != 0);
+
+  self->version = version;
+
+  g_object_notify_by_pspec (G_OBJECT (self), properties[PROP_VERSION]);
+}
+
+void
+modulemd_eol_set_modified (ModulemdEol *self, guint64 modified)
+{
+  g_return_if_fail (MODULEMD_IS_EOL (self));
+
+  self->modified = modified;
+
+  g_object_notify_by_pspec (G_OBJECT (self), properties[PROP_MODIFIED]);
+}
+
+
+void
+modulemd_eol_set_reset (ModulemdEol *self, gboolean reset)
+{
+  g_return_if_fail (MODULEMD_IS_EOL (self));
+  g_return_if_fail (!(reset && self->obsoleted_by_module_name));
+  g_return_if_fail (!(reset && self->obsoleted_by_module_stream));
+  g_return_if_fail (!(reset && self->eol_date));
```

Comment:
`g_return_if_fail()` is intended to catch coding errors (it calls `g_abort()` in the debug profile). Don't perform such checks in the set function anyway; the reason we have the validate function is to catch exactly these sorts of issues once all the data is loaded.